### DISCUSSION TITLE
not able to import autosar module

### DISCUSSION THREAD

**Discussion by krishomprasad** [cogu/autosar#126] [Q&A] [answered]
I installed autosar package 0,5,4 when I try to run the "run_examples.cmd" I get error saying "module not found: Autosar.xml" how shall I resolve this ?

**Comment by krishomprasad** [comment]
not sure what wrong am I doing here ? I can work with autosar package 0.4.1 which had setup.py but not with 0.5.4

**Reply by cogu** [reply]
On your machine, run `pip list` (or `pip3 list` in case your default pip is still on Python2).
That will tell you what version of the module `autosar` you have installed. If `autosar` is not even among the printed result then you have not installed it.

Please follow the README on the main page for correct installation instructions. Using virtual environment is optional.

**Reply by cogu** [reply]
I just checked the README and the install command is.
```bash
pip install  .
```
I fail to understand how this can go wrong for you. Unless of course your pip is still on python2.
You can always use the alternative `python -m pip` or `python3 -m pip` as replacement for the pip command.

**Comment by krishomprasad** [comment]
I get below error now 
![image](https://github.com/user-attachments/assets/79df5659-d4f9-4b39-9489-0032f71b8a07)

**Reply by cogu** [reply]
The `autosar.workspace` function only exists in v0.4.x. You can't run old scripts with the new version installed.

Are you aware of "Important Note 1" at the beginning of the project README?

`Python AUTOSAR v0.5+ uses a new API and is incompatible with earlier versions.`

Have you looked at any of the new example code under [autosar/examples](https://github.com/cogu/autosar/tree/master/examples/xml)?

**Reply by krishomprasad** [reply, marked as answer]
Yes , I did not read that but I understood that after the errors I got am able to run example scripts and will contact for any other queries , thank you !

**Comment by krishomprasad** [comment]
In the pip list I do see autosar installed 
![image](https://github.com/user-attachments/assets/735e4188-7745-4df0-9949-0baebd5b5b00)
